packages/sewing-kit-plugin-quilt/src/web-app-multi-build.ts

```diff
@@ -0,0 +1,188 @@
+  const browserGroups = Array.isArray(explicitBrowserGroups)
+    ? explicitBrowserGroups
+    : Object.keys(explicitBrowserGroups);
+  const browserGroupMap = new Map<string, string | string[]>();
+
+  if (!Array.isArray(explicitBrowserGroups)) {
+    for (const group of browserGroups) {
+      browserGroupMap.set(group, explicitBrowserGroups[group]);
+    }
+  }
+
+  return createProjectPlugin<WebApp>(PLUGIN, ({project, workspace, tasks}) => {
+    tasks.build.hook(({hooks}) => {
+      hooks.targets.hook(targets =>
+        targets.map(target =>
+          target.default && !target.runtime.includes(Runtime.Node)
+            ? target.multiply(currentTarget =>
+                ['default', ...browserGroups].map(browsers => ({
+                  ...currentTarget,
+                  browsers: browsers as BuildWebAppTargetOptions['browsers'],
+                })),
+              )
+            : target,
+        ),
+      );
+
+      hooks.configureHooks.hook(
+        addHooks<CustomHooks>(() => ({
+          quiltBrowserslist: new WaterfallHook(),
+        })),
+      );
+
+      hooks.target.hook(({target, hooks}) => {
+        hooks.configure.hook(configuration => {
+          const {browsers} = target.options;
+
+          configuration.webpackOutputDirectory?.hook(() => {
+            const topLevelPath = target.options.quiltAutoServer
+              ? 'server'
+              : 'assets';
+
+            return workspace.fs.buildPath(
+              workspace.webApps.length > 1 ? `apps/${project.name}` : 'app',
+              topLevelPath,
+            );
+          });
+
+          const id = idFromTargetOptions(target.options);
+
+          const addTargetIdToExtension = (filename: string) => {
+            return filename.replace(
+              /\.\w+$/,
+              extension => `${id ? `.${id}` : ''}${extension}`,
+            );
+          };
+
+          configuration.webpackOutputFilename?.hook(addTargetIdToExtension);
+          configuration.webpackOutputChunkFilename?.hook(
+            addTargetIdToExtension,
+          );
+          configuration.cssWebpackFileName?.hook(addTargetIdToExtension);
+
+          if (browsers == null) return;
+
+          configuration.webpackPlugins?.hook(async plugins => {
+            const [{AssetMetadataPlugin}] = await Promise.all([
+              import('@shopify/webpack-asset-metadata-plugin'),
+              configuration.quiltBrowserslist!.run([]),
+            ]);
+
+            return [
+              ...plugins,
+              new AssetMetadataPlugin(
+                idFromTargetOptions(target.options) as any,
+              ),
+            ];
+          });
+
+          configuration.quiltBrowserslist?.hook(async () => {
+            if (browserGroupMap.has(browsers)) {
+              return browserGroupMap.get(browsers);
+            }
+
+            const {default: browserslist} = await import('browserslist');
+            const browserslistResult = browserslist(undefined, {
```

Comment:
Why do we need to resolve this here? I think most tools (babel certainly) can target a browsers list expression rather than needing an explicit list of browsers